test/org/opensolaris/opengrok/web/api/v1/controller/MessagesControllerTest.java

```diff
@@ -186,4 +199,42 @@ private void setText(final Message m, final String text) throws Exception {
         f.set(m, text);
     }
 
+    @Test
+    public void getAllMessagesTest() {
+        addMessage("text1", "info");
+        addMessage("text2", "main");
+
+        List<AcceptedMessageModel> allMessages = target("messages")
+                .request()
+                .get(messagesType);
+
+        assertEquals(2, allMessages.size());
+    }
+
+    @Test
+    public void getSpecificMessageTest() {
+        addMessage("text", "info");
+
+        List<AcceptedMessageModel> messages = target("messages")
+                .queryParam("tag", "info")
+                .request()
+                .get(messagesType);
+
+        assertEquals(1, messages.size());
+        assertEquals("text", messages.get(0).message.getText());
+
+        assertThat(messages.get(0).message.getTags(), contains("info"));
+    }
+
+    @Test
+    public void multipleTagsTest() {
+        addMessage("test", "info", "main");
```

Comment:
I would also add a test for two messages, both multiple distinct tags, and selecting one tag and expecting one message in the output.